RSS Block: Add border-box CSS

scruffian opened the pull request:
## Description
I think we should add `box-sizing: border-box` to the RSS block. The block has a fairly large amount of padding. If we set the block to be full width with `width: 100%` then we'll get horizontal scrollbars on the page because of the padding. This CSS will fix that.

## How has this been tested?
In TT1 Blocks

## Types of changes
Breaking change - this could potentially break existing implementations, although I think it is more likely to fix them.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

scruffian committed:
RSS Block: Add border-box CSS

jasmussen commented:
This is a cool PR, but I'm seeing some weirdness that is probably unrelated to this PR, but also some that is. Screenshots after:

<img width="967" alt="Screenshot 2020-12-17 at 08 53 41" src="https://user-images.githubusercontent.com/1204802/102459409-d8d9b700-4045-11eb-8203-8d163a00c6d5.png">

<img width="834" alt="Screenshot 2020-12-17 at 08 53 57" src="https://user-images.githubusercontent.com/1204802/102459412-daa37a80-4045-11eb-9223-892135fc7715.png">

Frontend after:

<img width="847" alt="Screenshot 2020-12-17 at 08 54 06" src="https://user-images.githubusercontent.com/1204802/102459431-decf9800-4045-11eb-9c84-8af3309a643d.png">

Before is the same, actually:

<img width="804" alt="Screenshot 2020-12-17 at 08 54 59" src="https://user-images.githubusercontent.com/1204802/102459460-e8f19680-4045-11eb-9ed6-f00fdcd7b08f.png">

It appears there's a ton of padding on the RSS feed block caused by this:

```
.wp-block-rss {
    padding-left: 2.5em;
}
```

that appears to affect both the placeholder state, and the rendered state, but only in the editor.

<img width="792" alt="Screenshot 2020-12-17 at 08 55 32" src="https://user-images.githubusercontent.com/1204802/102459611-1e967f80-4046-11eb-8d96-c4df6fffae08.png">

Furthermore, in the editor the box-sizing property appears overridden by this:

<img width="367" alt="Screenshot 2020-12-17 at 08 56 21" src="https://user-images.githubusercontent.com/1204802/102459657-2ce49b80-4046-11eb-8615-5bfcdfecfb54.png">

I appreciate trying to keep the PR small, and I think it would be fine to land this. But since you're here — are you seeing the same issues, and should we look to fix them at the same time?

scruffian committed:
Remove padding rules and strengthen the box-sizing property

scruffian commented:
You're right, I removed the padding and added an `!important` to strengthen the border-box. There's a lot more we could do here, but I think it should happen as part of https://github.com/WordPress/gutenberg/issues/27766

jasmussen commented:
This is cool:

<img width="951" alt="Screenshot 2020-12-17 at 13 25 43" src="https://user-images.githubusercontent.com/1204802/102487785-67acfa80-406b-11eb-99c7-ebe4d9361ab3.png">

<img width="826" alt="Screenshot 2020-12-17 at 13 25 49" src="https://user-images.githubusercontent.com/1204802/102487790-68de2780-406b-11eb-98d2-ae7fa139b1f3.png">

But the need for an `!important` bugs me, and it seems to highlight an issue with the reset mixin having too high specificity. 

This seems to accomplish the same:

```
.wp-block-rss {

	// This needs extra specificity due to the reset mixin on the parent: https://github.com/WordPress/gutenberg/blob/a250e9e5fe00dd5195624f96a3d924e7078951c3/packages/edit-post/src/style.scss#L54
	&.wp-block-rss {
		box-sizing: border-box;
	}
```

👆 it will need a linter rule opt-out because the linter will say something about duplicate rules, I can't recall which rule. But it would elevate the specificity without going nuclear. 

What do you think?

scruffian commented:
Yeah I thought about that option but I thought that generally people would be more familiar with the `!important`. I agree that it's not ideal, but I don't see the double selector as a much better alternative. I suppose it does have the benefit of being more easily modified by the theme.

 Another option could be to modify the mixin so it doesn't target the rss block - `*:not(.wp-block-rss)` but that also feels bad.

It feels like we need a way to gradually move towards setting border-box on everything, but we've quite aggressively prevented that in the past, so it's going to be a long road!

jasmussen commented:
The problem with !important is that it's extremely hard to override, whereas the other approach adds just a little specificity, so I personally much prefer the other approach. 

But in order to not be blocked on this, I'm happy to defer to a sanity check from @kjellr or @MaggieCabrera.

scruffian committed:
Use a different approach for selector specificity

MaggieCabrera commented on packages/block-library/src/rss/style.scss:
```diff
@@ -1,5 +1,8 @@
-	box-sizing: border-box !important; // The important is needed because of the reset mixin on the parent: https://github.com/WordPress/gutenberg/blob/a250e9e5fe00dd5195624f96a3d924e7078951c3/packages/edit-post/src/style.scss#L54
+	// This needs extra specificity due to the reset mixin on the parent: https://github.com/WordPress/gutenberg/blob/a250e9e5fe00dd5195624f96a3d924e7078951c3/packages/edit-post/src/style.scss#L54
+	&.wp-block-rss {
+		box-sizing: border-box;
```
I prefer this much more than !important. I wonder why the theme css is not loaded after the editor's, wouldn't that solve all these kinds of issues?

kjellr replied:
Yeah, I prefer the double-selector approach here too. 👍

youknowriad replied:
It seems the extra specificity is only needed in the frontend, so I wonder whether we should keep "style" pure and add the specificity in "editor.scss". Not a big deal though.

scruffian replied:
IMO the benefits of having the same code in the editor and frontend are greater so I'd prefer to keep it like this.

jasmussen reviewed (approved):
Let's get this in. Thanks Ben.
